### PR TITLE
Do not create the service isolate when Dart is built in product mode

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

File: sky/engine/core/script/dart_init.cc [hunk 1]
```diff
@@ -131,6 +131,52 @@ bool IsServiceIsolateURL(const char* url_name) {
          String(url_name) == DART_VM_SERVICE_ISOLATE_NAME;
 }
 
+#ifdef FLUTTER_PRODUCT_MODE
```
Comment (Hixie) [inline sky/engine/core/script/dart_init.cc]: shouldn't this be release mode?

**Comment by Hixie** [comment]
Do we have a profile mode yet? Otherwise, this will break yegor's automated testing.

**Comment by Hixie** [comment]
LGTM

File: sky/engine/core/script/dart_init.cc (hunk 1 above)
Comment (jason-simmons) [inline sky/engine/core/script/dart_init.cc]: This is tied to Dart's notion of "product mode", which disables features such as the observatory

**Comment by jason-simmons** [comment]
Right now we aren't fully migrated to debug/profile/release modes because AOT isn't ready.  Flutter's current "release mode" is still using JIT and does not enable Dart product mode

**Comment by Hixie** [comment]
ok. LGTM. I think it's going to be confusing to have all these different mode names, though.

**Pull request merged**